core/new-gui/src/app/workspace/component/result-panel/result-panel.component.ts

```diff
@@ -305,6 +311,44 @@ export class ResultPanelComponent {
     this.workflowWebsocketService.send('ResultPaginationRequest', {pageSize: newPageSize, pageIndex: newPageIndex});
   }
 
+  public isHighlightedOperatorSink(): boolean {
+    const sinkOperator = this.getCurrentSinkOperator();
+    return sinkOperator && sinkOperator.operatorType === FILE_SINK_OP_TYPE;
+  }
+
+  public isOperatorExecuted(): boolean{
+    const executionState = this.executeWorkflowService.getExecutionState();
+    return executionState.state === ExecutionState.Completed &&
+    !!this.resultPanelOperatorID &&
+    executionState.resultMap.has(this.resultPanelOperatorID);
+  }
+
+  public downloadFile(){
+    const sinkOperator = this.getCurrentSinkOperator()
+    if (sinkOperator.operatorType !== FILE_SINK_OP_TYPE){
+      return;
+    } else if (!this.userService.isLogin()){
+      alert('Must login to download the result');
+      return;
+    }
```

Comment:
same comment, we might want to make it independent from the user system. use `texera-user` for default user.